### PR TITLE
Add default note to POS

### PR DESCRIPTION
**Pull request by ademar111190**
Fixes https://github.com/breez/breezmobile/issues/903

|Default|Custom|Cart|
|---|---|---|
|Screenshot_20220506-115458_Breez_Debug|Screenshot_20220506-115443_Breez_Debug|Screenshot_20220506-115429_Breez_Debug|

### CODE REVIEW

File: lib/bloc/pos_catalog/bloc.dart [hunk 1]
```diff
@@ -176,6 +180,16 @@ class PosCatalogBloc with AsyncActionsHandler {
     });
   }
 
+  void _trackDefaultSaleNote(Stream<BreezUserModel> userStream) {
+    userStream.listen((user) {
+      final defaultSaleNote = user.defaultPosNote;
+      final currentSale = _currentSaleController.value;
+      _currentSaleController.add(currentSale.copyWith(
```
Comment (roeierez) [inline lib/bloc/pos_catalog/bloc.dart:187]: just want to make sure that we don't override existing sale models. Where is the point that the _currentSaleController cleans up existing sales?

Comment (ademar111190) [reply lib/bloc/pos_catalog/bloc.dart:187]: It seems to be here: https://github.com/breez/breezmobile/blob/master/lib/bloc/pos_catalog/bloc.dart#L150

Comment (roeierez) [reply lib/bloc/pos_catalog/bloc.dart:187]: @ademar111190 what happens if the sale is not paid (e.g cancelled)? does the controller still hold the old value?

Comment (ademar111190) [reply lib/bloc/pos_catalog/bloc.dart:187]: If I cancel the value is kept, but if I clear the sale the value is set to empty 🤔 
I guess it should be "cleared" to the default right?

Comment (ademar111190) [reply lib/bloc/pos_catalog/bloc.dart:191]: I did apply this change, now the sale has the default note after a clear.

**Review by roeierez — APPROVED** [review]

**Pull request merged**